src/SqlProcessor.php

```diff
@@ -475,4 +483,21 @@ protected function getVariableTypeName($value)
 		return is_object($value) ? get_class($value) : (is_float($value) && !is_finite($value) ? $value : gettype($value));
 	}
 
+
+	/**
+	 * Invoked from processModifier.
+	 * Overrides existing modifiers.
+	 *
+	 * @param string   $type
+	 * @param callable $handler (string $type, mixed  $value)
+	 */
+	public function registerModifierProcessor($type, callable $handler)
+	{
+		if (!is_string($type) && !is_int($type)) {
```

Comment:
I probably just got carried away and should either check for is_string (which will translate to php 7 correctly) or check for scalar only.